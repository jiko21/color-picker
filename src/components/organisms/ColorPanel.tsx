import { ColorFrame } from '@/components/atoms/ColorFrame';
import { ColorForm } from '@/components/molecules/ColorForm';
import { Flex, View } from '@adobe/react-spectrum';
import React from 'react';

interface Props {
  color: string;
  setColor: (value: string) => void;
}

export const ColorPanel: React.FC<Props> = React.memo(({ color, setColor }) => {
  const handleChange = (value: string) => {
    setColor(value);
  };

  return (
    <ColorFrame color={color} width="100%" height="100%">
      <Flex width="100%" height="100%" direction="column" alignItems="center" justifyContent="center">
        <View>
          <Flex direction="row" alignItems="center" justifyContent="center" gap="size-100">
            <ColorForm color={color} handleChange={handleChange} />
          </Flex>
        </View>
      </Flex>
    </ColorFrame>
  );
});
