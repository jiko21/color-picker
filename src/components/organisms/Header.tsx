import React from 'react';
import { Flex, Header as SpectrumHeader, View } from '@adobe/react-spectrum';

interface HeaderProps {
  children: React.ReactNode;
}

export const Header: React.FC<HeaderProps> = ({ children }) => {
  return (
    <SpectrumHeader>
      <Flex direction="row" gap="size-400" alignItems="center" justifyContent="center">
        <View>{children}</View>
      </Flex>
    </SpectrumHeader>
  );
};
