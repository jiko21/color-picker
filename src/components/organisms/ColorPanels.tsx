import { ColorPanel } from '@/components/organisms/ColorPanel';
import { useMobileDevice } from '@/hooks/useMobileDevice';
import { copyUrlToClipboard } from '@/util/url';
import { ActionButton, Flex, Tooltip, TooltipTrigger, View } from '@adobe/react-spectrum';
import Add from '@spectrum-icons/workflow/Add';
import Link from '@spectrum-icons/workflow/Link';
import React from 'react';

interface ColorPanelsProps {
  colors: string[];
  addColor: () => void;
  setColor: (value: string, i: number) => void;
}

export const ColorPanels: React.FC<ColorPanelsProps> = ({ colors, addColor, setColor }) => {
  const clickLink = () => {
    copyUrlToClipboard();
  };

  const isMobile = useMobileDevice();

  const direction = isMobile ? 'column' : 'row';

  return (
    <Flex direction={direction} width="100%" height="calc(100vh - size-200)" gap="size-10">
       <View width="100%" height="100%">
        <Flex direction={direction} width="100%" height="100%">
           {colors.map((color, i) => (
            <ColorPanel color={color} setColor={(value) => setColor(value, i)} key={i} />
           ))}
        </Flex>
        <View position="absolute" bottom="size-500" right="size-500" zIndex={100}>
          <TooltipTrigger>
            <ActionButton aria-label="Link Only" onPress={clickLink} data-testid="link-button">
              <Link aria-label="LinkIcon" />
            </ActionButton>
            <Tooltip>Share this color schema with link</Tooltip>
          </TooltipTrigger>
        </View>
       </View>
       <View alignSelf="center" justifySelf="center">
        <ActionButton aria-label="Icon Only" onPress={addColor} data-testid="add-button">
          <Add aria-label="AddIcon" />
        </ActionButton>
       </View>
    </Flex>
  );
};
