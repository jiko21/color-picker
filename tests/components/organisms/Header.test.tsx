import { Header } from '@/components/organisms/Header';
import { render } from '@testing-library/react';
import React from 'react';

describe('components/Organisms/Header.tsx', () => {
  it('correctly render', () => {
    const component = render(<Header>Sample</Header>);
    expect(component.container).toMatchSnapshot();
  });
});
