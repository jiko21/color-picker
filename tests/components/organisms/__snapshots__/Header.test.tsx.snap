// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Organisms/Header.tsx correctly render 1`] = `
<div>
  <header>
    <div
      class="flex-container_e15493"
    >
      <div
        class="flex_e15493 flex-gap_e15493"
        style="flex-direction: row; align-items: center; justify-content: center; --gap: var(--spectrum-global-dimension-size-400, var(--spectrum-alias-size-400));"
      >
        <div>
          Sample
        </div>
      </div>
    </div>
  </header>
</div>
`;
